frontend/src/heat_map/democracy_viz.js

```diff
@@ -0,0 +1,351 @@
+                    <li>
+                        The country does exist but there is no data for that particular
+                        score type in that particular year
+                    </li>
+                </ul>
+                <br/>
+                <DemocracyMap
+                    democracyData={this.state.democracyData}
+                    scoreType={this.state.scoreType}
+                    year={this.state.year}
+                />
+            </>
+        )
+    }
+}
+
+const scoreTypeRanges = {
+    "v2x_polyarchy": {
+        "min": 0,
+        "max": 1
+    },
+    "v2x_partipdem": {
+        "min": 0,
+        "max": 1
+    },
+    "v2x_freexp_altinf": { //ordinal
+        "min": 0,
+        "max": 1
+    },
+    "v2x_cspart": {
+        "min": 0,
+        "max": 1
+    },
+    "v2xel_locelec": {
+        "min": 0,
+        "max": 1
+    },
+    "v2elboycot": {
+        // what does ordinal values mean and how to convert it to
+        // what is in the database
+        "min": 0,
+        "max": 1
+    },
+    "v2lpname": { // this is a text value???
+        "min": 0,
+        "max": 1
+    },
+    "v2slpname": { // text value
+        "min": 0,
+        "max": 1
+    },
+    "v2tlpname": { //text value
+        "min": 0,
+        "max": 1
+    },
+    "v2psnatpar": { //ordinal
+        "min": 0,
+        "max": 1
+    },
+    "v2excrptps": { //ordinal
+        "min": 0,
+        "max": 1
+    },
+    "v2juaccnt": { //ordinal
+        "min": 0,
+        "max": 1
+    },
+    "v2svstterr": { //percentage
+        "min": 0,
+        "max": 100
+    },
+    "v2meharjrn": { //ordinal
+        "min": 0,
+        "max": 1
+    },
+    "v2peprisch": { //percentage
+        "min": 0,
+        "max": 100
+    },
+    "v2pesecsch": {
+        "min": 0,
+        "max": 100
+    },
+    "v2smonex": { //ordinal
+        "min": 0,
+        "max": 1
+    },
+    "e_migdppc": { //GDP so it can be anything
+        "min": 0,
+        "max": 60000
+    },
+    "e_mipopula": { //population
+        "min": 0,
+        "max": 100000
+    },
+    "gdpcapl": { //not in guidebook
+        "min": 0,
+        "max": 60000
+    },
+    "ethfrac": { //not in guidebook
+        "min": 0,
+        "max": 1
+    },
+    "relfrac": { //not in guidebook
+        "min": 0,
+        "max": 1
+    },
+    "numlang": { //not in guidebook
+        "min": 0,
+        "max": 1
+    },
+    "colbrit": { //not in guidebook
+        "min": 0,
+        "max": 1
+    },
+    "colfra": { //not in guidebook
+        "min": 0,
+        "max": 1
+    },
+    "gdpcap_currusd": { //not in guidebook
+        "min": 0,
+        "max": 1
+    },
+    "adultliteracy": { //not in guidebook
+        "min": 0,
+        "max": 1
+    },
+    "elecaccessrur": { //not in guidebook
+        "min": 0,
+        "max": 1
+    },
+};
+
+export class DemocracyMap extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            lng: 29,
```

Comment:
there's some cruft leftover here from map quiz that we can remove now, so we don't let it build up!